tests/scsi/007

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+# SPDX-License-Identifier: GPL-3.0+
+# Copyright 2021 Google LLC
+#
+# Trigger the SCSI error handler.
+
+. tests/scsi/rc
+. common/scsi_debug
+
+DESCRIPTION="Trigger the SCSI error handler"
+
+QUICK=1
+
+requires() {
+	_have_scsi_debug
+}
+
+group_device_requires() {
+	true
+}
+
+config_hz() {
+	if [ -e /proc/config.gz ]; then
+		zcat /proc/config.gz
+	else
+		cat "/boot/config-$(uname -r)"
+	fi | sed -n 's/^CONFIG_HZ=//p'
+}
+
+trigger_scsi_eh() {
+	if ! _init_scsi_debug ""; then
+		return
+	fi
+
+	# Enable SCSI error handler logging
+	echo 63 > /sys/module/scsi_mod/parameters/scsi_logging_level
+
+	local dev freq jdelay target
+	for target in "${SCSI_DEBUG_TARGETS[@]}"; do
+		if ! dev=$(cd "/sys/class/scsi_device/${target}/device/block" &&
+			echo *); then
+			echo "Error: scsi_debug instance not found"
+			continue
+		fi
+		# Change the block layer timeout to max(1 / CONFIG_HZ, 0.001)
+		# seconds.
+		echo 1 > "/sys/class/block/$dev/queue/io_timeout"
+		echo "I/O timeout = $(<"/sys/class/block/$dev/queue/io_timeout")" >>"$FULL"
+		# Change the scsi_debug delay to 5 seconds.
+		freq=$(config_hz)
+		jdelay=$((5 * "${freq}"))
+		echo "CONFIG_HZ=${freq} jdelay=${jdelay}" >>"$FULL"
+		echo "$jdelay" > /sys/module/scsi_debug/parameters/delay
+		{
+			if dd if="/dev/$dev" of=/dev/null bs=512 count=1 \
+				iflag=direct >&/dev/null; then
+				echo "Reading from scsi_debug succeeded"
+		  	else
+				echo "Reading from scsi_debug failed"
+			fi
+		} &
+		sleep 5
+		echo "Restoring I/O timeout"
+		# Change the block layer timeout to 1 second.
+		echo 1000 > "/sys/class/block/$dev/queue/io_timeout"
+		wait
+	done
+
+	# Disable SCSI error handler logging
+	echo 0 > /sys/module/scsi_mod/parameters/scsi_logging_level
+
+	echo "Test complete"
+
+	_exit_scsi_debug
+}
+
+test() {
+	echo "Running ${TEST_NAME}"
+
+	trigger_scsi_eh
```

Comment:
Just move the contents of `trigger_scsi_eh` into here and get rid of that function.

Reply:
Done.